ecosystem/node-checker/src/metric_collector/reqwest_metric_collector.rs

```diff
@@ -79,6 +76,25 @@ impl MetricCollector for ReqwestMetricCollector {
             .await
             .with_context(|| format!("Failed to process response body from {}", url))
             .map_err(|e| MetricCollectorError::ResponseParseError(anyhow!(e)))?;
+        Ok(body)
+    }
+}
+
+#[async_trait]
+impl MetricCollector for ReqwestMetricCollector {
+    async fn collect_metrics(&self) -> Result<Vec<String>, MetricCollectorError> {
+        let body = self.get_data_from_node("metrics").await?;
```

Comment:
nit: we may want to make "metrics" and "system_information" constants. Thankfully they shouldn't change after this: https://github.com/aptos-labs/aptos-core/pull/1439, but I'll eventually work on a constants crate we can share for metrics and endpoint names, etc.